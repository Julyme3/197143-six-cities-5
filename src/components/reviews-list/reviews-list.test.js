import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list";

const reviews = [
  {
    id: 2,
    date: `2020-11-05T13:38:44.753Z`,
    src: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
    name: `Christina`,
    raiting: 4,
    text: `I stayed here for one night and it was an unpleasant experience.`,
    dateFormated: `November 2020`,
  },
  {
    id: 3,
    date: `2020-10-05T13:38:44.753Z`,
    src: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/12.jpg`,
    name: `Julia`,
    raiting: 2.8,
    text: `The room was spacious and clean.`,
    dateFormated: `October 2020`,
  },
];

test(`ReviewsList render`, () => {
  const tree = renderer
    .create(
        <ReviewsList
          reviews={reviews}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
