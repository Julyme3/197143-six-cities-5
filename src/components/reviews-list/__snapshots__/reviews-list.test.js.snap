// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList render 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Christina
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        I stayed here for one night and it was an unpleasant experience.
      </p>
      <time
        className="reviews__time"
        dateTime="November 2020"
      >
        November 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/12.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Julia
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "56%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        The room was spacious and clean.
      </p>
      <time
        className="reviews__time"
        dateTime="October 2020"
      >
        October 2020
      </time>
    </div>
  </li>
</ul>
`;
